tests/common/fixtures/snappi_fixtures.py

```diff
@@ -0,0 +1,413 @@
+        snappi_api_serv_port (pytest fixture): snappi_api_serv_port fixture.
+    """
+    host = "https://" + snappi_api_serv_ip + ":" + str(snappi_api_serv_port)
+    # TODO: Currently extension is defaulted to ixnetwork.
+    # Going forward, we should be able to specify extension
+    # from command line while running pytest.
+    api = snappi.api(host=host, ext="ixnetwork")
+
+    yield api
+
+    if getattr(api, 'assistant', None) is not None:
+        api.assistant.Session.remove()
+
+
+def __gen_mac(id):
+    """
+    Generate a MAC address
+    Args:
+        id (int): IXIA port ID
+    Returns:
+        MAC address (string)
+    """
+    return '00:11:22:33:44:{:02d}'.format(id)
+
+
+def __valid_ipv4_addr(ip):
+    """
+    Determine if a input string is a valid IPv4 address
+    Args:
+        ip (unicode str): input IP address
+    Returns:
+        True if the input is a valid IPv4 adress or False otherwise
+    """
+    try:
+        return True if type(ip_address(ip)) is IPv4Address else False
+    except ValueError:
+        return False
+
+
+def __l3_intf_config(config, port_config_list, duthost, ixia_ports):
+    """
+    Generate Tgen configuration of layer 3 interfaces
+    Args:
+        config (obj): Tgen API config of the testbed
+        port_config_list (list): list of IXIA port configuration information
+        duthost (object): device under test
+        ixia_ports (list): list of IXIA port information
+    Returns:
+        True if we successfully generate configuration or False
+    """
+    mg_facts = duthost.minigraph_facts(host=duthost.hostname)['ansible_facts']
+    if 'minigraph_interfaces' in mg_facts:
+        l3_intf_facts = mg_facts['minigraph_interfaces']
+    else:
+        return True
+
+    if len(l3_intf_facts) == 0:
+        return True
+
+    l3_intf = {}
+    for v in l3_intf_facts:
+        if __valid_ipv4_addr(v['addr']):
+            l3_intf[v['attachto']] = v
+
+    dut_mac = str(duthost.facts['router_mac'])
+
+    for k, v in l3_intf.items():
+        intf = str(k)
+        gw_addr = str(v['addr'])
+        prefix = str(v['prefixlen'])
+        ip = str(v['peer_addr'])
+
+        port_ids = [id for id, ixia_pot in enumerate(ixia_ports) \
+                    if ixia_pot['peer_port'] == intf]
+        if len(port_ids) != 1:
+            return False
+
+        port_id = port_ids[0]
+        mac = __gen_mac(port_id)
+
+        device = config.devices.device(
+            name='Device Port {}'.format(port_id),
+            container_name=config.ports[port_id].name)[-1]
+
+        ethernet = device.ethernet
+        ethernet.name = 'Ethernet Port {}'.format(port_id)
+        ethernet.mac = mac
+
+        ip_stack = ethernet.ipv4
+        ip_stack.name = 'Ipv4 Port {}'.format(port_id)
+        ip_stack.address = ip
+        ip_stack.prefix = prefix
+        ip_stack.gateway = gw_addr
+
+        port_config = IxiaPortConfig(id=port_id,
```

Comment:
IxiaPortConfig -> SnappiPortConfig

Reply:
Created new file in snappi folder and updated the class name